src/unix/thread.c

```diff
@@ -85,29 +86,30 @@ int pthread_barrier_wait(pthread_barrier_t* barrier) {
     return rc;
 
   /* Increment the count. If this is the first thread to reach the threshold,
-     wake up waiters, unlock the mutex, then return
-     PTHREAD_BARRIER_SERIAL_THREAD. */
+     wake up waiters, unlock the mutex. */
   if (++barrier->in == barrier->threshold) {
     barrier->in = 0;
     barrier->out = barrier->threshold - 1;
     rc = pthread_cond_signal(&barrier->cond);
     assert(rc == 0);
 
     pthread_mutex_unlock(&barrier->mutex);
-    return PTHREAD_BARRIER_SERIAL_THREAD;
+    return 0;
```

Comment:
In a 1-thread barrier, shouldn't this return `PTHREAD_BARRIER_SERIAL_THREAD`?

Reply:
👍Fixed.